R/client.R

```diff
@@ -15,16 +15,30 @@
 emodnet_init_wfs_client <- function(service, service_version = "2.0.0") {
 
     service <- match.arg(service, choices = EMODnetWFS::emodnet_wfs$service_name)
+    service_url <- get_service_url(service)
 
-    wfs <- suppressWarnings(ows4R::WFSClient$new(get_service_url(service),
-                         serviceVersion = service_version))
+    create_client <- function(){
+
+        wfs <- suppressWarnings(ows4R::WFSClient$new(service_url,
+                                                     serviceVersion = service_version))
+
+        check_wfs(wfs)
+        usethis::ui_done("WFS client created succesfully")
+        usethis::ui_info("Service: {usethis::ui_value(wfs$getUrl())}")
+        usethis::ui_info("Version: {usethis::ui_value(wfs$getVersion())}")
+
+        wfs
+    }
```

Comment:
I personally don't often contain the definition of functions within another function, generally preferring for separate functions and explicitly specify arguments and pass them during execution (which `create_client` currently doesn't do). However, that doesn't mean I'm right! and as R will look for `service_url ` and `service_version ` in the parent environment in which the inputs have already been checked, I guess there's no danger of bugginess with the approach here. 

Was just wondering whether there was a specific reason you went for this approach that I'm not aware of? @maelle do you have any strong thoughts on this topic?

Reply:
I don't have any strong thought, it seems fine to have it inside the function if that's the only place where it's used. :-)

Reply:
I tried with including this whole code inside the `tryCatch()` but it failed. It seemed to run only the first line. Before looking too much into the problem I just tried this way (putting the code into a function and call it in the tryCatch) and it worked fine :-)

Reply:
All good @salvafern , it makes sense and works fine so no need to change 😄